include/mysql/service_encryption.h

```diff
@@ -114,11 +116,16 @@ static inline int encryption_crypt(const unsigned char* src, unsigned int slen,
 {
   void *ctx= alloca(encryption_ctx_size(key_id, key_version));
   int res1, res2;
-  unsigned int d1, d2;
+  unsigned int d1, d2= *dlen;
+
+  assert(*dlen >= slen);
```

Comment:
I'd also add something like
```
  assert(dst[*dlen-1]=1);
```
which _could_ crash and _will_ trigger asan/msan/valgrind if `*dlen` is too large